src/toDoList.js

```diff
@@ -2,39 +2,66 @@ import React from 'react';
 
 class ToDoList extends React.Component {
   constructor() {
-      super();
-      this.state = {
+      super(); //it calls react Component constructor.
+      this.state = { //state is a plain js object..init only in constructor
         items: [],
         value: ''
       };
       this.addItem = this.addItem.bind(this);
       this.displayList = this.displayList.bind(this);
+      this.removeItem = this.removeItem.bind(this);
   }
 
   addItem() {
-    let inputText = this.state.value;
-    this.state.items = [...this.state.items, inputText];
-    this.state.value = '';
+    let inputText = this.state.value; //object destructring
+    let newItems = [...this.state.items, inputText];
+    //this.state.value = '';  dont change state variables directly... use setState instead
     this.setState({
       value : '',
-      items : this.state.items
+      items : newItems
     });
   }
 
-  displayList(){
-    return (this.state.items.slice(0).reverse().map((item,i) => <li key={i}>{item}</li>));
+  removeItem(id) {
+    let newItems = this.state.items.filter((item, i) => i !== id);
+    this.setState({
+      items : newItems
+    });
+  }
+
+  displayList() {
+    return (<ul>{this.state.items.map((item, i) => {
```

Comment:
code after return should be on new line

Reply:
done - formatted code in toDoItem.js after refactoring